src/Sulu/Bundle/AdminBundle/Resources/js/containers/Badge/stores/BadgeStore.js

```diff
@@ -0,0 +1,122 @@
+import jsonPointer from 'json-pointer';
+import debounce from 'debounce';
+import symfonyRouting from 'fos-jsrouting/router';
+import Router from '../../../services/Router';
+import Requester from '../../../services/Requester';
+import type {HandleResponseHook} from '../../../services/Requester/types';
+
+export default class BadgeStore {
+    router: Router;
+    routeName: string;
+    dataPath: ?string;
+    requestParameters: Object;
+    routerAttributesToRequest: Object;
+    @observable value: ?string = null;
+    disposer: () => {};
+
+    constructor(
+        router: Router,
+        routeName: string,
+        dataPath: ?string,
+        requestParameters: Object,
+        routerAttributesToRequest: Object
+    ) {
+        this.router = router;
+        this.routeName = routeName;
+        this.dataPath = dataPath;
+        this.requestParameters = requestParameters;
+        this.routerAttributesToRequest = routerAttributesToRequest;
+    }
+
+    @computed get evaluatedRequestParameters() {
+        const {
+            router: {
+                attributes: routerAttributes,
+            },
+            requestParameters: attributesToRequest,
+            routerAttributesToRequest,
+        } = this;
+
+        const requestParameters = {};
+        Object.keys(routerAttributesToRequest)
+            .forEach((routerAttributeKey) => {
+                const requestAttributeKey = routerAttributesToRequest[routerAttributeKey];
+                const attributeName = isNaN(routerAttributeKey)
+                    ? routerAttributeKey
+                    : requestAttributeKey;
+
+                requestParameters[requestAttributeKey] = routerAttributes[attributeName];
+            });
+
+        return {...requestParameters, ...attributesToRequest};
+    }
+
+    @computed get url() {
+        const {routeName} = this;
+
+        return symfonyRouting.generate(routeName, this.evaluatedRequestParameters);
+    }
+
+    @action setData(data: any) {
+        const {dataPath} = this;
+
+        let enhancedData = data;
+        if (dataPath) {
+            enhancedData = jsonPointer.get(data, dataPath);
+        }
+
+        this.value = String(enhancedData);
+    }
+
+    load = debounce(() => {
+        Requester.get(this.url).then((response: Object) => {
+            this.setData(response);
+        });
+    }, 3000, true);
+
+    responseHook: HandleResponseHook = (response: Response, options: ?Object) => {
+        if (!options || typeof options.method === 'undefined') {
+            return;
+        }
+
+        if (response.url.includes(this.url)) {
+            return;
+        }
+
+        if (response.url.includes('/admin/api/collaborations')) {
+            return;
+        }
+
+        if (response.url.includes('/admin/preview/')) {
+            return;
+        }
+
+        if (['POST', 'PUT', 'PATCH', 'DELETE'].includes(options.method.toUpperCase())) {
+            this.load();
+        }
+    };
+
+    initialize = () => {
```

Comment:
any reason for not doing this in the constructor? did we do it like this in other places too? 🙂

Reply:
I think you said you would register the `HandleResponseHook` in the `componentDidMount` ... Should i only keep that in the `initialize()` method and keep the rest in the `constructor()`